### DISCUSSION TITLE
Unwanted dependency between Meshery adapters and Kubernetes.

### DISCUSSION THREAD

**Discussion by leecalcote** [meshery/meshery#2591] [General]
There is an unwanted dependency between Meshery adapters and Kubernetes. When Meshery Server is not connected to Kubernetes, the ad-hoc ping tests to check connectivity with the adapters fails. This means that without a Kubernetes connection users see Meshery Adapters as being unavailable (disconnected from Meshery Server). 

A fix for this can be to leave the dependency more or less in place, but change the ad-hoc ping test that is performed when a user clicks on the chip in the UI.

[Proposal] Have two tests run:
1. Connectivity test between Meshery Server and Meshery Adapter.
1. Connectivity test between Meshery Adapter and Kubernetes.

**Comment by kumarabd** [comment]
A simple fix here would be to ignore errors while running `Create Instance` call during initialisation. @mgfeller @utkarsh-pro sounds about right?

**Reply by mgfeller** [reply]
Hmm... I assume this is the handler that is used: AdapterPingHandler, correct?

For pinging, an adapter->k8s connection isn't really needed, as mentioned, is that correct? 
Just checking that the adapter is reachable?
If so, instead of creating a grpc client including creating the connection to from the adapter to k8s,
something like this should suffice, shouldn't it (from `meshery/meshes/client.go:CreateClient`)?

```
	opts = append(opts, grpc.WithInsecure())
	// }
	conn, err := grpc.Dial(meshLocationURL, opts...)
	if err != nil {
		logrus.Errorf("fail to dial: %v", err)
	}

```

i.e. replace the relevant code in AdapterPingHandler with this, rather than ignoring errors? It isn't quite clear to me which errors to ignore (I can see some candidates), and this approach might require some logic that might appear illogical :smile: 
imho, changing the code as mentioned appears to be clearer :smile:

**Reply by mgfeller** [reply]
I wondered, isn't successful dialing a prerequisite, i.e. it should return an error?
One could add a parameter "pingOnly" or similar to CreateClient and ignore the error from` mClient.CreateMeshInstance`, just log it, yes.

**Comment by leecalcote** [comment]
// @vijcheru

**Comment by leecalcote** [comment]
@kumarabd an item to address as you attach kubeconfig concerns in Meshery Server now.
